store/splus.js

```diff
@@ -1,28 +1,101 @@
+import Vue from 'vue';
+import colors from 'vuetify/es5/util/colors';
+import * as moment from 'moment';
 import SCHEDULES from '~/assets/schedules.json';
 
+const hashCode = (string) =>
+string.split('').reduce((prevHash, currVal) =>
+  (((prevHash << 5) - prevHash) + currVal.charCodeAt(0))|0, 0);
+
+
 export const state = () => ({
-  lectures: [],
+  schedule: SCHEDULES[0],
   schedules: SCHEDULES,
+  /**
+   * Map of { week: lectures[] }
+   */
+  lectures: {},
+  /**
+   * Currently viewed iso week of year.
+   * Week 53 of year 2018 equals week 1 of year 2019.
+   */
+  week: moment().isoWeek(), // TODO won't work in 2019
 });
 
 export const getters = {
-  getLecturesByWeekAndSchedule: (state) => (week, schedule) => {
-    return state.lectures.filter((lecture) =>
-      lecture.week == week && lecture.schedule == schedule);
+  getHasLecturesOnWeekend: (state) => {
+    if (state.lectures[state.week] == undefined) {
+      return false;
+    }
+
+    return state.lectures[state.week].filter(lecture => lecture.day > 5).length > 0;
+  },
+  /**
+   * @return The lectures as timestamp-aware dayspan calendar event inputs.
+   * @see https://clickermonkey.github.io/dayspan/docs/interfaces/eventinput.html
+   */
+  getLecturesAsEvents: (state) => {
+    if (state.lectures[state.week] == undefined) {
+      return [];
+    }
+
+    const colorsArr = Object.values(colors).slice(0, -1); // exclude black
+
+    return state.lectures[state.week].map((lecture) => {
+      const beginHours = Math.floor(lecture.begin);
+      const start = moment()
+        .isoWeek(lecture.week)
+        .day(lecture.day + 1)
+        .hour(beginHours)
+        .minute((lecture.begin - beginHours) * 60);
+      const hashOfFirstWordInTitle = hashCode(lecture.title.split(' ')[0]) + Math.pow(2, 31);
+      const color = colorsArr[hashOfFirstWordInTitle % colorsArr.length].lighten1;
+
+      return {
+        data: {
+          title: lecture.title,
+          color,
+          description: `${lecture.lecturer} ${lecture.room} ${lecture.info}`,
+          location: lecture.room,
+        },
+        schedule: {
+          on: start,
+          times: [ {
+            hour: start.hour(),
+            minute: start.minute(),
+          } ],
+          duration: lecture.end - lecture.begin,
+          durationUnit: 'hours',
+        }
+      };
+    });
   },
 };
 
 export const mutations = {
-  addLectures(state, lectures) {
-    lectures
-      .filter((lecture) => !state.lectures.find((lecture2) => lecture.id == lecture2.id))
-      .forEach((lecture) => state.lectures.push(lecture));
+  addLectures(state, { lectures, week }) {
+    // reactive variant of state.lectures[week] = lectures
+    Vue.set(state.lectures, week, lectures);
```

Comment:
Was bringt ist dann reactive?

Reply:
Werden dann Änderungen der lectures erkannt?

Reply:
Must read: https://vuejs.org/v2/guide/reactivity.html

Vue beobachtet die Daten im state (bzw. `data` in Komponenten). Wenn die sich ändern, werden die getter (bzw. computed properties) neu berechnet und ggf. das Layout neu gerendert.
Die Beobachtung funktioniert nur, wenn sich das ganze Objekt ändert. `state.lectures = {}` wird als Änderung erkannt, `state.lectures.foo = 123` nicht. Als "hint" gibt es `Vue.set`.
In Angular gibt es diese Problematik übrigens auch.